### PR TITLE
feat: Adds pagination and date to service journey/departure details

### PR DESCRIPTION
**Pull request by mikaelbr**
This adds pagination and date metadata to departure details view.

- Unifies behavior/design for trip pattern details and service journey/departure details.
- Adds disappearing map on scroll.
- Adds pagination for service journey/departure details
- Delays loading indicator 400 ms to avoid it showing too often.

## Examples

Screenshot 2021-01-27 at 15 23 19
Screenshot 2021-01-27 at 15 23 49

Screenshot 2021-01-27 at 15 23 32

## Demo new scroll behavior for details view

https://user-images.githubusercontent.com/606374/106045112-deb0c680-60e0-11eb-915d-af69434e2257.mov

Closing #879

### CODE REVIEW

File: src/components/disappearing-header/content.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  },
+}));
+
+function useCalculateHeaderContentHeight() {
+  const {onLayout: onHeaderContentLayout, height: contentHeight} = useLayout();
+  return {
+    contentHeight,
+    onHeaderContentLayout,
+  };
+}
```
Comment (cbrevik) [inline src/components/disappearing-header/content.tsx:159]: Maybe an unnecessary abstraction?

File: src/screens/Nearby/section-items/line.tsx [hunk 2]
```diff
@@ -72,29 +72,32 @@ export default function LineItem({
...
+  const onPress = (activeItemIndex: number) => {
     navigation.navigate('TripDetails', {
       screen: 'DepartureDetails',
       params: {
-        title,
-        serviceJourneyId: departure.serviceJourneyId!,
-        date: departure.aimedTime,
-        fromQuayId: group.lineInfo?.quayId,
+        activeItemIndex,
+        items,
```
Comment (cbrevik) [inline src/screens/Nearby/section-items/line.tsx:86]: Was thinking about how we'd link to "trips" with deep-links when we get as far as providing that functionality. But we probably need some other way of looking up trips when we get that far anyway. Not that sending in `items` breaks anything, still serializable, just thinking out loud.

**Review by cbrevik — COMMENTED** [review]
Tested the branch locally, and dark mode doesn't seem quite right. Unsure if related to this PR or not:
image

File: src/screens/Nearby/section-items/line.tsx (hunk 2 above)
Comment (mikaelbr) [reply src/screens/Nearby/section-items/line.tsx:86]: I think this still supports deep linking in the same way as before. Items is just references to service journeys which can be URLs

Comment (cbrevik) [reply src/screens/Nearby/section-items/line.tsx:86]: Yeah I think so too, just thinking out loud if this screen is maybe evolving past the scope of "linking to a single trip". If we ever get there that support can deep-link trips for users, or users can save trips. Not important wrt to this PR 👯

File: src/components/disappearing-header/content.tsx (hunk 1 above)
Comment (mikaelbr) [reply src/components/disappearing-header/content.tsx:159]: Agreed 😬 I initially planned to use this component inside the existing disappearing-header, but decided against it due to #746

**Comment by mikaelbr** [comment]
> Tested the branch locally, and dark mode doesn't seem quite right. Unsure if related to this PR or not:

Definitely related to my changes! Thanks 😬  Will look into it.

File: src/screens/Nearby/section-items/line.tsx (hunk 2 above)
Comment (mikaelbr) [reply src/screens/Nearby/section-items/line.tsx:86]: Note, this is for service journeys, not trips.

**Comment by cbrevik** [comment]
RefreshControl overlays header. But that's probably a good thing so navigation arrows don't move when you're trying to page through many trips?
image

**Review by cbrevik — APPROVED** [review]
LGTM, great job landing this!

**Comment by mikaelbr** [comment]
> RefreshControl overlays header. But that's probably a good thing so navigation arrows don't move when you're trying to page through many trips?

Yes, this is to avoid jumping content. Not perfect, and we can get a better showing of this, but it is an existing problem also so we can solve it in a separate issue

**Pull request merged**